forem/forem
# Enable webmentions

## What type of PR is this? (check all applicable)


## Description
This PR is based on a few assumptions that i thought would be enough to test webmentions.

Since there exists different ways to [discover webmentions](https://www.w3.org/TR/webmention/#sender-discovers-receiver-webmention-endpoint), this PR assumes that the `canonical_url` points to a URL that advertises its webmention endpoint with an HTML <link> tag in the document and that it ~~should be an absolute URL~~ (i added relative URL support)

Also, Since DEV will be the source of webmentions, these have to be in a certain format a.k.a `Microformats` that allows the receiving website to parse various information of your posts like the comment's content, the author of the comment's information(name, picture, etc..)

More info can be found here:
- [Sending your first webmention](https://aaronparecki.com/2018/06/30/11/your-first-webmention)
- [Microformats Wiki](http://microformats.org/wiki/h-entry)

## Related Tickets & Documents
Closes #6807 
## QA Instructions, Screenshots, Recordings

- Create or edit a post and add `canonical_url` that supports Webmentions(i.e https://webmention.rocks/test/4)
- Write a comment on that post and it should be handled by the Webmention server(i.e shown on https://webmention.rocks/test/4)

FYI: For this to work locally, and be able to send post requests to the Webmention endpoint, I had to:
- Use`ngrok` and change `ApplicationConfig["APP_DOMAIN"]` to `ngrok` URL
- Configure the `development` environment to allow requests from the `ngrok` URL

## Added tests?


## Added to documentation?


![](https://media.giphy.com/media/26tP8qAV3GyP1Pdde/giphy.gif)
